### ISSUE TITLE
The workbox-webpack-plugin generate an error url in the precache-manifest.js file

### ISSUE THREAD

**Issue by pfdgithub** [GoogleChrome/workbox#1493]
**Library Affected**:
workbox-webpack-plugin  

**Browser & Platform**:
windows 10 x64  
node.js 8.11.1  
webpack 4.8.3  
workbox-webpack-plugin 3.2.0  
html-webpack-plugin 3.2.0  

**Issue or Feature Request Description**:
Use ```webpack``` to generate the ```js``` file in the ```/dist/assets/``` directory.  
Use ```html-webpack-plugin``` to generate the ```html``` file in the ```/dist/``` directory.  
The ```workbox-webpack-plugin``` will generate an error ```url``` in the ```precache-manifest.js``` file.  

Here is an example https://github.com/pfdgithub/workbox-webpack-plugin-demo  

webpack.config.js  
```
const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const WorkboxPlugin = require('workbox-webpack-plugin');

module.exports = {
  mode: 'development',
  entry: {
    index: './src/index.js'
  },
  output: {
    path: path.resolve(__dirname, 'dist', 'assets'),
    filename: '[name].[chunkhash].js',
    publicPath: '/assets/',
  },
  plugins: [
    new HtmlWebpackPlugin({
      filename: path.resolve(__dirname, 'dist', 'index.html')
    }),
    new WorkboxPlugin.GenerateSW({
      swDest: path.resolve(__dirname, 'dist', 'service-worker.js')
    })
  ]
};
```

Build output
```
Hash: 6f0da71c3b43dd24faf2
Version: webpack 4.8.3
Time: 598ms
Built at: 2018-05-24 11:18:28
                                                Asset       Size  Chunks             Chunk Names
                        index.0c3568d41c904e638ac1.js   3.14 KiB   index  [emitted]  index
                                        ..\index.html  210 bytes          [emitted]
precache-manifest.1b5cad419bfeb0a74d42b688531989eb.js  186 bytes          [emitted]
                                 ..\service-worker.js  915 bytes          [emitted]
Entrypoint index = index.0c3568d41c904e638ac1.js
[./src/index.js] 309 bytes {index} [built]
Child html-webpack-plugin for "..\index.html":
     1 asset
    Entrypoint undefined = ..\index.html
    [./node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 509 bytes {0} [built]
    [./node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 519 bytes {0} [built]
        + 2 hidden modules
```

precache-manifest.js  
```
self.__precacheManifest = [
  {
    "url": "/assets/index.0c3568d41c904e638ac1.js"
  },
  {
    "revision": "1e49e2609410a3c4b3a3f57f23d50519",
    "url": "/assets/..\\index.html"
  }
];
```

**Comment by pfdgithub** [comment]
Not just html-webpack-plugin, there are other plugins like [webpack-pwa-manifest](https://github.com/arthurbergmz/webpack-pwa-manifest).  
Even with the right webpack configuration, these plugins still output the wrong asset name in windows, like this issues https://github.com/arthurbergmz/webpack-pwa-manifest/issues/65.  

Can workbox-webpack-plugin provide a configuration like [manifestTransforms](https://developers.google.com/web/tools/workbox/modules/workbox-webpack-plugin#full_generatesw_config), allow developers to correct the wrong asset name?

**Comment by jeffposnick** [comment]
This should be addressed by the current Workbox v5.0.0 alpha.
